genesyscloud/routing_sms_addresses/resource_genesyscloud_routing_sms_addresses.go

```diff
@@ -0,0 +1,155 @@
+package genesyscloud
+
+import (
+	"context"
+	"fmt"
+	"log"
+	gcloud "terraform-provider-genesyscloud/genesyscloud"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v105/platformclientv2"
+)
+
+const resourceName = "genesyscloud_routing_sms_address"
+
+func getAllRoutingSmsAddress(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(resourceExporter.ResourceIDMetaMap)
+	proxy := getRoutingSmsAddressProxy(clientConfig)
+
+	allSmsAddresses, err := proxy.getAllSmsAddresses(ctx)
+	if err != nil {
+		return nil, diag.Errorf("failed to get sms addresses: %v", err)
+	}
+
+	for _, entity := range *allSmsAddresses {
+		var name string
+		if entity.Name != nil {
+			name = *entity.Name
+		} else {
+			name = *entity.Id
+		}
```

Comment:
This is the bug fix. Before, if `Name` wasn't returned in an entity object, the same stack track from the ticket description would appear.

Reply:
Nice